docs/code-i18n.md

```diff
@@ -0,0 +1,31 @@
+# Code Internationalization
+
+When a project is intended to be localized to multiple languages, all the strings in the codebase must also be translated. Any language strings that are "hardcoded" in PHP or JavaScript files must follow the code internationalization practices to ensure they can be translated into any language you want to support.
+
+This is done by passing all strings that your custom code outputs through the translation API functions. This is used to both swap your strings for their translated variant at runtime depending on the language of the site, and also be used as markers to generating translation template files (POT files).
+
+Once you have generated a POT file for you custom code, you can provide your translators with these files. Typically programs such as [poedit](https://poedit.net/) are used to author translations.
+
+## Translation API Functions
+
+Whenever you output a string in code, you must use a translation function. HM Platform uses the [WordPress Translation API](https://codex.wordpress.org/I18n_for_WordPress_Developers) to provide code level internationalization (i18n). Translations in code are tied to a "Text Domain" which is an identifier for which translation file should be used for a given translation. It's recommended you use a single translation domain for all your project's custom code. This will mean you only need to get a single POT file translated.
+
+See [Translatable Strings](https://codex.wordpress.org/I18n_for_WordPress_Developers#Translatable_strings) for details on the functions you should use, depending on context.
+
+## Generating POT files
+
+Once you have ensured all the strings in your custom code are using translation functions, you must generated a POT file from the codebase. This is done use the cli command `wp i18n make-pot`. If you want to generate a POT file for your whole project:
+
+```sh
+wp i18n make-pot content --domain=<your-text-domain> <your-text-domain>.pot
```

Comment:
Might need a little work still but we could bundle either this WP CLI command or one that I started at https://github.com/roborourke/wp-l10n-gen

My one has a bit more flexibility in terms of conversions and output types plus excluding directories like `node_modules` and `vendor`. It can be used to generate `.po` files for each language and then convert them once they've been filled in.

Will be nice to add a build script example though just realised there's no WP CLI during build. Maybe we can make the GNU gettext command line tools available on the build server.

Reply:
Hmm, why would one run make-pot at build time? You'd then need to somehow get it from the build server and download it anyway

Reply:
Sorry, to clarify that would be to make `.mo` files from `.po` files at build time but possibly kinda pointless at that stage.